distributed/shuffle/_shuffle.py

```diff
@@ -140,6 +150,11 @@ def rearrange_by_column_p2p(
         name,
         meta,
         [None] * (npartitions + 1),
+    ).map_partitions(
+        load_output_partition,
+        layer._tokens[1],
```

Comment:
@hendrikmakait - This is the shuffle-barrier key. Sorry for the hacky nature of this line (it would have been a miracle for you to see this on your own).